src/test/java/core/basesyntax/service/strategy/impl/PurchaseOperationHandlerTest.java

```diff
@@ -0,0 +1,103 @@
+package core.basesyntax.service.strategy.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.MapStorage;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitService;
+import core.basesyntax.service.impl.FruitServiceImpl;
+import core.basesyntax.service.impl.FruitTransactionProcessorImpl;
+import core.basesyntax.service.strategy.OperationHandlerStrategy;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class PurchaseOperationHandlerTest {
+    private static final String TEST_FRUIT_NAME = "watermelon";
+    private static final int TEST_QUANTITY = 50;
+    private static FruitTransactionProcessorImpl fruitTransactionProcessor;
+    private static Storage storage;
+
+    @BeforeAll
+    static void setUp() {
+        storage = new MapStorage();
+        FruitService fruitService = new FruitServiceImpl(storage);
+        OperationHandlerStrategy strategy = new OperationHandlerStrategyImpl(
+                Map.of(FruitTransaction.Operation.BALANCE,
+                        new BalanceOperationHandler(fruitService),
+                        FruitTransaction.Operation.PURCHASE,
+                        new PurchaseOperationHandler(fruitService)
+                )
+        );
+        fruitTransactionProcessor = new FruitTransactionProcessorImpl(
+                strategy
+        );
+    }
+
+    @AfterEach
+    void afterEach() {
+        MapStorage mapStorage = (MapStorage) storage;
+        mapStorage.clear();

```

Comment:
Why we need to create an addition variable for storage if we already have one?
```suggestion
       storage.clear();
```
Fix all similar cases in all test classes.